tips/TIP-0020/tip-0020.md

```diff
@@ -0,0 +1,501 @@
+                  <tr>
+                    <td><b>Name</b></td>
+                    <td><b>Type</b></td>
+                    <td><b>Description</b></td>
+                  </tr>
+                  <tr>
+                    <td>Input Type</td>
+                    <td>uint8</td>
+                    <td>
+                      Set to <strong>value 0</strong> to denote an <i>UTXO Input</i>.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td>Transaction ID</td>
+                    <td>ByteArray[32]</td>
+                    <td>The BLAKE2b-256 hash of the transaction payload containing the referenced output.</td>
+                  </tr>
+                  <tr>
+                    <td>Transaction Output Index</td>
+                    <td>uint16</td>
+                    <td>The output index of the referenced output.</td>
+                  </tr>
+                </table>
+              </details>
+            </td>
+          </tr>
+          <tr>
+            <td>Inputs Commitment</td>
+            <td>ByteArray[32]</td>
+            <td>
+              BLAKE2b-256 hash of the BLAKE2b-256 hashes of the serialized outputs referenced in <i>Inputs</i> by their <i>Output IDs</i> (<i>Transaction ID</i> || <i>Transaction Output Index</i>).
+            </td>
+          </tr>
+          <tr>
+            <td>Outputs Count</td>
+            <td>uint16</td>
+            <td>The number of output entries.</td>
+          </tr>
+          <tr>
+            <td valign="top">Outputs <code>anyOf</code></td>
+            <td colspan="2">
+              <details>
+                <summary>Basic Output</summary>
+                <blockquote>
+                  Describes a deposit to a single address. The output might contain optional features and native tokens.
+                </blockquote>
+              </details>
+              <details>
+                <summary>Alias Output</summary>
+                <blockquote>
+                  Describes an alias account in the ledger.
+                </blockquote>
+              </details>
+              <details>
+                <summary>Foundry Output</summary>
+                <blockquote>
+                  Describes a foundry that controls supply of native tokens.
+                </blockquote>
+              </details>
+              <details>
+                <summary>NFT Output</summary>
+                <blockquote>
+                  Describes a unique, non-fungible token deposit to a single address.
+                </blockquote>
+              </details>
+            </td>
+          </tr>
+          <tr>
+            <td>Payload Length</td>
+            <td>uint32</td>
+            <td>The length in bytes of the optional payload.</td>
+          </tr>
+          <tr>
+            <td valign="top">Payload <code>optOneOf</code></td>
+            <td colspan="2">
+              <details>
+                <summary>Generic Payload</summary>
```

Comment:
Is the payload really generic?
Even though that we are defining the allowed payloads in the following paragraph, it differs from the way how the Outputs are defined above for example. Could be misleading IMO.